transformations/contextual_meaning_perturbation/README.md

```diff
@@ -0,0 +1,49 @@
+# Contextual Meaning Perturbation🦎  + ☎️️ → 🐍
+This transformation changes the meaning of the sentence while avoiding grammar, spelling and logical mistakes.
+
+
+## What type of a transformation is this?
+This transformation's effect is comparable to the <i>"Chinese Whispers"</i> or <i>"Telephone"</i> children's game: The transformed sentence
+appears fluent and somewhat logical, but the meaning of the original sentence is not preserved.
+To achieve logical coherence, a pre-trained language model is used to replace words with alternatives that match the context of the sentence.
+Grammar mistakes are reduced by limiting the type of words considered for changes (based on POS tagging) and replacing adjectives with adjectives,
+nouns with nouns, etc. where possible.
+
+
+## What tasks does it intend to benefit?
+This transformation benefits users who seek perturbations that preserve fluency but not the meaning of the sentence.
+For instance, it can be used in scenarios where the meaning is relevant to the task, but the model shows a tendency
+to over-rely on simpler features such as the grammatical correctness and general coherence of the sentence.
+A real-world example would be the training of quality estimation models for machine translation
+(does the translation maintain the meaning of the source?) or for text summarisation (does the summary capture the content of the source?).
+
+
+
+## What kind of configurations are tweakable?
+- <b>pos_to_change</b>: Determines which tokens are eligible for change based on the part of speech – default: ['ADV','ADJ', 'VERB', 'NOUN', 'PROPN']
+- <b>perturbation_rate</b>: The percentage of eligible tokens that should be changed – default: 0.3
+- <b>language</b>: Currently supported languages include English (en) and German (de) – default: "en"
+- <b>top_k</b>: Determines how many replacement candidates should be proposed – default: 10
+
```

Comment:
Could you add the results of evaluate.py on the tasks intended to be addressed by the transformation?

Reply:
Hi @mmeyer39, my transformation is mostly intended to mitigate partial-input biases for quality estimation tasks, as an addition to a labelled quality estimation dataset. I could evaluate against the sentence classification tasks, but I don't think this would add much value in the README. I added information on the behaviour of the transformation when using the pre-configured parameters – I believe that might be more helpful than reporting benchmark results. What's your opinion on that? Many thanks! :)

Reply:
Hi, 
I mentioned it because it was asked in the email from the organizers. However, it is okay as such for me :)